controllers/endpoint/endpoints_controller.go

```diff
@@ -40,7 +40,7 @@ type Reconciler struct {
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := logf.FromContext(ctx)
 	//Get the endpoint object
-	var ep v1.Endpoints
+	var ep discoveryv1.EndpointSlice
 	err := r.Client.Get(ctx, req.NamespacedName, &ep)
```

Comment:
you can consider making a small change here:
```go
if err := r.Client.Get(ctx, req.NamespacedName, &epSlice); err != nil {
  return ctrl.Result{}, err
}
```
There is no need to use RequeueAfter when also returning an error. Let it use the exponential backoff in case of errors.

Reply:
Fixed with cf53853fa3d67b0ecf7908dcbba24f95952ea7a9